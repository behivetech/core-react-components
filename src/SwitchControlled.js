// Vendor Libs
import React, {Component} from 'react';

// Components
import Switch from './Switch';
import withFormControl from './withFormControl';
import withSwitchControl from './withSwitchControl';

class SwitchControlled extends Component {
	render() {
		return <Switch {...this.props} className="checkbox-controlled" />;
	}
}

export default withFormControl(withSwitchControl(SwitchControlled));