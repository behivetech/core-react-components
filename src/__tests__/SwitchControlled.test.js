import SwitchControlled from '../SwitchControlled';

jest.mock('../withFormControl');
jest.mock('../withSwitchControl');

describe('SwitchControlled', () => {
    it('should render with the correct props', () => {
        const wrapper = shallow(<SwitchControlled name="mockName" className="mock-class" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.dive()).toMatchSnapshot();
        expect(wrapper.dive().dive()).toMatchSnapshot();
    });
});