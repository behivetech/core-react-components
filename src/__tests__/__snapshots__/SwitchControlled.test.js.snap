// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchControlled should render with the correct props 1`] = `
<SwitchControlled
  className="mock-class"
  formState={
    Object {
      "getFieldError": [MockFunction],
      "getFieldValue": [MockFunction],
      "setFieldValidators": [MockFunction],
      "setFieldValue": [MockFunction],
      "setFieldValueDebounced": [MockFunction],
      "shouldDisable": [MockFunction],
    }
  }
  name="mockName"
/>
`;

exports[`SwitchControlled should render with the correct props 2`] = `
<SwitchControlled
  className="mock-class"
  name="mockName"
/>
`;

exports[`SwitchControlled should render with the correct props 3`] = `
<Switch
  checked={false}
  className="checkbox-controlled"
  name="mockName"
  onChange={[Function]}
  value={true}
/>
`;
